### PR TITLE
Draft: Feature/good protocolUI-210: Add savings

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

This PR is for adding the Savings feature and implementing web3sdk-v2

About # (link your issue here)
#210

# How Has This Been Tested?
Locally on fuse env (desktop)

# Additional context (screens)
## DESKTOP
(StakeDeposit - main)
Copy headers:
- [ ] token - This is the token that is currently being staked.
- [ ] protocol - This is the protocol the token is staked to.
- [ ] fixed apy - The fixed annual interest.
- [ ] total staked - Total currently saved.
- [ ] total rewards paid -  Total rewards claimed.

image

(StakeDeposit - deposit modal)

image

(stakeDeposit - deposit transaction send)

image

(StakeDeposit - deposit success)
- [ ] !Copy required: What do we want to show on deposit, or keep it general?)
image

(Portfolio - main)
Copy headers:
- [ ] Type - {copy here}
- [ ] Token - This is the token that is currently being staked.
- [ ] Protocol - This is the protocol that the token staked to.
- [ ] Deposit - The total of your deposits which accumulates the rewards.
- [ ] Claimable rewards -  How much tokens your deposits have accumulated so far
image

(Portfolio - claim rewards start)
- [ ] Is Copy needed? - {copy here}
image

(Portfolio - claim rewards send)

image

(Portfolio - claim rewards success)
- [ ] !Copy required: {copy here}

image

(Portfolio - Withdraw)
- [ ] !Disclaimer required: Withdraw also claims rewards {copy here}
image

(Portfolio - Withdraw progress)
- [ ] Copy needed? : {copy here}
image

(Portfolio - Withdraw success) 
- [ ] ! Copy required: {copy here}
image

## Mobile
(Stake)
image

(Portfolio)
image

### CODE REVIEW

**Comment by L03TJ3** [comment]
GA Events:
image

**Comment by sirpy** [comment]
@L03TJ3 the savings buttons are not aligned to the right end like the buttons above
in general try to make sure the other components have same alignment, margins,paddings etc

**Comment by sirpy** [comment]
@L03TJ3 remove yalc and add it to ignore

File: src/components/gd/Button/ActionOrSwitchButton.tsx [hunk 1]
```diff
@@ -30,11 +31,13 @@ export const ActionOrSwitchButton = ({
...
     const toggleNetworkModal = useNetworkModalToggle()
     const { i18n } = useLingui()
-    const { chainId, account } = useActiveWeb3React()
+    const { chainId } = useActiveWeb3React()
     // useEffect(() => { }, [chainId, account])
     if (
-        (chainId === (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.FUSE) ||
-        (chainId !== (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.MAINNET)
+        (chainId === (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.FUSE ) ||
+        (chainId !== (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.MAINNET) ||
```
Comment (sirpy) [inline src/components/gd/Button/ActionOrSwitchButton.tsx:38]: this line will be true for CELO.
this logic can probably be simplified, by converting requireNetwork to the relevant chainIds
DAO_NETWORK.FUSE = 122
DAO_NETWORK.CELO = 42220
DAO_NETWORK.MAINNET = 1,3,42
then check `ids.includes(chainId)`

File: src/pages/gd/Stake/index.tsx [hunk 2]
```diff
@@ -483,8 +483,8 @@ export default function Stakes(): JSX.Element | null {
                     )}
                 </Modal>
             </StakesSC>
-            {network !== 'production' ? 
-              <Savings network={DAO_NETWORK.FUSE} chainId={chainId} /> : <></>
+            {network !== 'production' && library ? 
+              <Savings network={network} chainId={chainId} /> : <></>
```
Comment (sirpy) [inline src/pages/gd/Stake/index.tsx:487]: i've asked that you dont pass network or chainid because you can get them via hooks in Savings

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/components/gd/Button/ActionOrSwitchButton.tsx (hunk 1 above)
Comment (L03TJ3) [reply src/components/gd/Button/ActionOrSwitchButton.tsx]: As explained in slack, this button is for a action on a specific network/chain.
so I believe your proposed solution wouldn't work.
But I do think I have now committed a simplified version

File: src/pages/gd/Stake/index.tsx (hunk 2 above)
Comment (L03TJ3) [reply src/pages/gd/Stake/index.tsx]: added hook for network

Comment (sirpy) [reply src/pages/gd/Stake/index.tsx]: you still keep passing network to other sub components why?

Comment (L03TJ3) [reply src/pages/gd/Stake/index.tsx]: has been removed now, moved to be handled in sdk

**Comment by sirpy** [comment]
@L03TJ3 please review my last commit and make sure it works as expected.
specifically the "stake" page that is supposed to display data correctly regardless of the network the user is on
and on "portfolio" page that is supposed to show user data when he is connected

**Comment by L03TJ3** [comment]
note: latest commit works only with mono local sdkv1/v2 from branch getnetworkenv_and_usecalls_fix

**Comment by L03TJ3** [comment]
The finalizations of copy, and UI updates will be done through a new issue.

**Pull request merged**